### DISCUSSION TITLE
Google access

### DISCUSSION THREAD

**Discussion by radar335** [derekhe/msfs2020-google-map#139] [Q&A]
Tried this program yesterday - and it worked like a charm.  Very impressing. I closed it properly, and checked the hosts file to ensure the two code lines was deleted. Today it didn't work at all.  When testing the connection in the program panel it says: "Google access failed"
It seems like Google is blocked. The strange thing is that it seems to also affect other services that use google maps. I use flightradar24.com who uses google maps as backgroud, but that map is also gone. Is it possible my PC altogether or my IP is blocked entirely? Or maybe an certificate issue?

**Comment by Marakai** [comment]
I have had this happen as well and my suspicion is actually throttling on the Google side :( While I can't easily prove it, Google may not like the sheer number of requests coming in from one IP address. If you watch the logs, there are a LOT coming in!

I'm looking at the code and wondering whether it might not be better to make it use the various Google maps servers in a round-robin fashion, so requests are spread out so to speak.

**Reply by radar335** [reply]
I think you're absolutely right. Afer I wrote this ticket, I tested with my laptop, and the same had happened there. (maps gone for other applikcations). While my iPad running on the 4G-network was ok. Therefore it seems like the trottle back is related to the IP- adress , not the actual PC. The day after the the maps were back, and it looks like the IP-adress is now "pleaded not guilty"

**Reply by Marakai** [reply]
Indeed. The author and I are testing some alternate hostnames to use (see my PR). While it seems simpler to just use the load-balancer hostname, there is still a risk of throttling in case the LB uses "session stickiness" always direct to the same underlying maps server. I've added some round-robin code in my PR, but ideally that would work on a known list of underlying maps hostnames - the more the better.
